files/jvm/src/main/java/com/swoval/files/FileCacheDirectoryTree.java

```diff
@@ -414,7 +415,12 @@ public void close() {
     final Path absolutePath = path.isAbsolute() ? path : path.toAbsolutePath();
     if (directoryRegistry.addDirectory(absolutePath, maxDepth) && directories.lock()) {
       try {
-        watcher.register(absolutePath, maxDepth);
+        final Either<IOException, Boolean> res = watcher.register(absolutePath, maxDepth);
```

Comment:
is it OK to swallow the exception here? As a user, I'd expect this method to throw if any failure occurred during registration given the method signature

Reply:
That's a good point. It probably should throw.